rpc/handlers.go

```diff
@@ -1463,14 +1463,24 @@ func (h *Handler) simulateTransactions(id BlockID, transactions []BroadcastedTra
 
 	var result []SimulatedTransaction
 	for i, overallFee := range overallFees {
+		feeUnit := feeUnit(txns[i])
+
+		gasPrice := header.GasPrice
+		if feeUnit == FRI {
+			gasPrice = header.GasPriceSTRK
+			if gasPrice == nil {
+				gasPrice = &felt.Zero
+			}
```

Comment:
Should this go outside the enclosing `if`?

Reply:
`GasPrice` is known to be non-nil in all blocks AFAIK